Global with auto increment will cause c error when instancing struct with reference to global.

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->

**V doctor:**

```
V full version: V 0.4.8 5c65e58.54a6915
OS: windows, Microsoft Windows 11 Pro v22631 64-bit
Processor: 24 cpus, 64bit, little endian, 

getwd: D:\vbug2
vexe: C:\v\v.exe
vexe mtime: 2024-10-30 07:36:18

vroot: OK, value: C:\v
VMODULES: OK, value: C:\Users\info\.vmodules
VTMP: OK, value: C:\Users\info\AppData\Local\Temp\v_0

Git version: git version 2.42.0.windows.2
Git vroot status: weekly.2024.44-1-g54a6915f-dirty
.git/config present: true

CC version: Error: 'cc' is not recognized as an internal or external command,

operable program or batch file.


thirdparty/tcc status: thirdparty-windows-amd64 b425ac82
```

**What did you do?**
`./v -g -o vdbg cmd/v && ./vdbg main.v`

```v
module main

struct Game {
mut:
	world  GameObject
	object_id u32
}

__global (
	game Game
)

struct GameObject {
	id int = game.object_id++
}

fn (mut game Game) init() {
	game.world = GameObject{}
}

fn main() {}
```

**What did you expect to see?**

No c compiler error

**What did you see instead?**

```
$ /c/v/v -enable-globals main.v
================== C compilation error (from tcc): ==============
cc: C:/Users/info/AppData/Local/Temp/v_0/main.01JBEJ2T39NEXA9C73XA9QXDNH.tmp.c:7046: warning: implicit declaration of function 'tcc_backtrace'
cc: C:/Users/info/AppData/Local/Temp/v_0/main.01JBEJ2T39NEXA9C73XA9QXDNH.tmp.c:13633: error: expected struct or union but not 'struct main__Game *'
... (the original output was 3 lines long, and was truncated to 2 lines)
=================================================================
(You can pass `-cg`, or `-show-c-output` as well, to print all the C error messages).
builder error:
==================
C error found. It should never happen, when compiling pure V code.
This is a V compiler bug, please report it using `v bug file.v`,
or goto https://github.com/vlang/v/issues/new/choose .
You can also use #help on Discord: https://discord.gg/vlang .
```

> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

<sub><a href="https://huly.app/guest/vlang-66f40c4d-a476b54c67-771fdd?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJsaW5rSWQiOiI2NzIyMTU2NjI3Y2M3NjgxYzYyM2E1N2EiLCJndWVzdCI6InRydWUiLCJlbWFpbCI6IiNndWVzdEBoYy5lbmdpbmVlcmluZyIsIndvcmtzcGFjZSI6InctYWxleGFuZGVyLXZsYW5nLTY2ZjQwYzRkLWE0NzZiNTRjNjctNzcxZmRjIn0.TbwIO_c_Y8V9ePilLMnIw38jhSLc8-yAopTwev28jGo">Huly&reg;: <b>V_0.6-21146</b></a></sub>

--- a/vlib/v/checker/fn.v
+++ b/vlib/v/checker/fn.v
@@ -192,6 +192,10 @@ fn (mut c Checker) fn_decl(mut node ast.FnDecl) {
 		}
 	}
 	if node.is_method {
+		if node.receiver.name in c.global_names {
+			c.error('cannot use global variable name `${node.receiver.name}` as receiver',
+				node.receiver_pos)
+		}
 		if node.receiver.typ.has_flag(.option) {
 			c.error('option types cannot have methods', node.receiver_pos)
 		}

--- a/vlib/v/checker/tests/globals/global_receiver_var_name_err.out
+++ b/vlib/v/checker/tests/globals/global_receiver_var_name_err.out
@@ -1,0 +1,7 @@
+vlib/v/checker/tests/globals/global_receiver_var_name_err.vv:17:9: error: cannot use global variable name `game` as receiver
+   15 | }
+   16 | 
+   17 | fn (mut game Game) init() {
+      |         ~~~~~~~~~
+   18 |     game.world = GameObject{}
+   19 | }

--- a/vlib/v/checker/tests/globals/global_receiver_var_name_err.vv
+++ b/vlib/v/checker/tests/globals/global_receiver_var_name_err.vv
@@ -1,0 +1,21 @@
+module main
+
+struct Game {
+mut:
+	world     GameObject
+	object_id u32
+}
+
+__global (
+	game Game
+)
+
+struct GameObject {
+	id int = game.object_id++
+}
+
+fn (mut game Game) init() {
+	game.world = GameObject{}
+}
+
+fn main() {}

